## **Getting Started with Python**

### **1. Hello World Program!**

In [1]:
# printing "Hello World" because its the rule!
print("Hello World!")

Hello World!


**Exploring More!**

In [2]:
print(20+50)

70


In [3]:
print('20+50')

20+50


> Because in `'' or ""` python recognize them as sentence (string). That's why we got as it is value in second time we run the cell `print('20+50')`

In [4]:
print(30>5)

True


In [5]:
print(6<5)

False


> Here, `30` is greater than 5 which is `True` thats why we got True as an output.

### **2. Variables and Data Types**

In [1]:
# creating a python variable is super easy, all we have to do is:
a = 15

**Rules To create a variable in python**

In [2]:
# variable name should not starts with number
1a = 15

SyntaxError: invalid decimal literal (2269455664.py, line 2)

In [3]:
a1 = 15

In [4]:
# variable name should not have space
a 2 = 10

SyntaxError: invalid syntax. Perhaps you forgot a comma? (746718547.py, line 2)

In [5]:
a_2 = 15

**Memory address alocation in python**

In [19]:
a = 10

In [20]:
print("Address of a = ",id(a))

Address of a =  1857617527312


In [21]:
b = a

In [22]:
print("Address of a = ",id(b))

Address of a =  1857617527312


In [23]:
print("Address of b = ",id(b))

Address of b =  1857617527312


In [24]:
print("Is addresses of both variables are same?",id(a)==id(b))

Is addresses of both variables are same? True


In [25]:
# lets change the value of variable a
a = 'Hello'
print(id(a))

1857698929456


**What I did here?**
> As you can see that the new value assigned to a different address which means that the `variable a` still holds the prev data in address = `1857617527312` and new data is stored in new address = `1857698929456`. Which means that this data type cannot be changed therefore we say that this is an **immutable variable**.

### **Data Types in Python**

1. **String Data Types**

In [26]:
varA = 'Hello this is an example of string data type (basically sentences or words)'

In [27]:
# To check type of a variable we use `type()` func
print('Type of varA = ',type(varA))

Type of varA =  <class 'str'>


> **One noteable thing is here that in python we don't have any data types , we have `data classes` instead.**

**Operation on strings**

In [28]:
# string concatination / addition of two or more strings
string1 = 'Hello baby '
string2 = 'How do you do'
print(string1 + string2)

Hello baby How do you do


In [29]:
# another way
print('Hello baby '+'How do you do')

Hello baby How do you do


In [30]:
# capitalization
print(string1.capitalize())

Hello baby 


In [32]:
# uppercase
print(string2.upper())

HOW DO YOU DO


In [33]:
# lower
print(string1.lower())

hello baby 


In [36]:
#  okay I will do it later